storage/source/storage/ChunkStore.cpp

```
@@ -583,6 +583,12 @@ std::pair<FhgfsOpsErr, int> ChunkStore::openAndChown(const int targetFD, const s
          return {FhgfsOpsErrTk::fromSysErr(errno), -1};
    }
 
+   int r = fcntl(fd, F_SET_RW_HINT, &writeHint);
+   if (r < 0) {
+      LOG(GENERAL, ERR, "Failed to set writeHint.",
+      ("writeHint", StringTk::uint64ToStr(writeHint)));
+   }
```

todo: If `writeHint == RWH_WRITE_LIFE_NOT_SET` (0), we should skip calling `fcntl(F_SET_RW_HINT, …)` entirely, since no hint was requested.

If a non-zero `writeHint` is provided and `fcntl()` fails for any reason (including EINVAL for unsupported kernels), we should log an error rather than silently dropping the hint. This makes it visible when the client expects lifetime hints to be applied but the server cannot honor them.